examples/end-to-end-examples/support_chatbot/chatbot.py

```diff
@@ -0,0 +1,546 @@
+    def __init__(self, 
+                 search_type: str,
+                 vector_store: FAISS,
+                 k: int,
+                 score_threshold: int):
+        self.k = k
+        self.vector_store=vector_store
+        self.score_threshold = score_threshold
+        self.search_type=search_type
+
+    def _get_relevant_documents(self, query: str) -> list[Document]:
+        # [NOTE] we removed the search type, only use search_type = "similarity"
+        if self.search_type != "similarity":
+            raise ValueError(f"Only search_type='similarity' is supported with scores")
+        docs_and_scores = self.vector_store.similarity_search_with_score(query=query, 
+                                                                         k=self.k, 
+                                                                         score_threshold=self.score_threshold)
+        for doc, score in docs_and_scores:
+            doc.metadata = {**doc.metadata, **{"score": score}}
+        return [doc for (doc, _) in docs_and_scores]
+
+    def aget_relevant_documents(self, query):
+        return self._get_relevant_documents(query)
+    
+    def get_relevant_documents(self, query: str) -> list[Document]:
+        return self._get_relevant_documents(query)
+
+__all__ = ['ChatBot']
+
+EVAL_7B_TEMPLATE = (f'Answer the following question as one function, class, or object. If you do not know, just say "I do not know".'
+                    '\n{context}'
+                    '\nQuestion: {question}')
+
+EVAL_30B_TEMPLATE = ("""<|im_start|>system
+                     A conversation between a user and an LLM-based AI assistant about the codebase for the MosaicML library Composer. 
+                     Provide a helpful and simple answer given the following context to the question. If you do not know, just say "I 
+                     do not know".<|im_end|>
+                     <|im_start|>context
+                     {context}<|im_end|>
+                     <|im_start|>user
+                     {question}<|im_end|>
+                     <|im_start|>assistant""")
+SUBQUERY_INTENT_TEMPLATE = ("""<|im_start|>system
+                               A conversation between a user and an LLM-based AI assistant about the codebase for MosaicML. 
+                               Provide a helpful, short and simple answer given the following context to the question. If you do not know, just say "I 
+                               do not know".<|im_end|>
+                               <|im_start|>context
+                               {context}<|im_end|>
+                               <|im_start|>user
+                               Why would the user ask the following question: {question}<|im_end|>
+                               <|im_start|>assistant""")
+PARTIAL_SUBQA_TEMPLATE = ("""<|im_start|>system
+                             A conversation between a user and an LLM-based AI assistant about the codebase for MosaicML. 
```

Comment:
this one might work better as two separate queries rather than the "if this then that, else this"